Modules/_multiprocessing/multiprocessing.c

```diff
@@ -179,25 +156,42 @@ PyInit__multiprocessing(void)
             py_sem_value_max = PyLong_FromLong(INT_MAX);
         else
             py_sem_value_max = PyLong_FromLong(SEM_VALUE_MAX);
-        if (py_sem_value_max == NULL)
-            return NULL;
+
+        if (py_sem_value_max == NULL) {
+            Py_DECREF(py_sem_value_max);
+            return -1;
+        }
         PyDict_SetItemString(_PyMp_SemLockType.tp_dict, "SEM_VALUE_MAX",
                              py_sem_value_max);
     }
-    PyModule_AddObject(module, "SemLock", (PyObject*)&_PyMp_SemLockType);
+
+    /* Add _PyMp_SemLock type to module */
+    Py_INCREF(&_PyMp_SemLockType);
+    if (PyModule_AddType(module, &_PyMp_SemLockType) < 0) {
+        Py_DECREF(&_PyMp_SemLockType);
+        return -1;
+    }
+
 #endif
 
     /* Add configuration macros */
-    temp = PyDict_New();
-    if (!temp)
-        return NULL;
+    PyObject *flags = PyDict_New();
+    if (!flags) {
+        return -1;
+    }
 
-#define ADD_FLAG(name)                                            \
-    value = Py_BuildValue("i", name);                             \
-    if (value == NULL) { Py_DECREF(temp); return NULL; }          \
-    if (PyDict_SetItemString(temp, #name, value) < 0) {           \
-        Py_DECREF(temp); Py_DECREF(value); return NULL; }         \
-    Py_DECREF(value)
+#define ADD_FLAG(name) {                                    \
```

Comment:
There are many compiler warnings/errors on this macro. I suggest:

```
#define ADD_FLAG(name)                                          \
    do {                                                        \
        PyObject *value = PyLong_FromLong(name);                \
        if (value == NULL) {                                    \
            Py_DECREF(flags);                                   \
            return -1;                                          \
        }                                                       \
        if (PyDict_SetItemString(flags, #name, value) < 0) {    \
            Py_DECREF(flags);                                   \
            Py_DECREF(value);                                   \
            return -1;                                          \
        }                                                       \
        Py_DECREF(value);                                       \
    } while (0)
```

Reply:
@vstinner I don't see any warnings or errors in my local windows build or the ubuntu CI. I'll change it as you suggest

Reply:
> @vstinner I don't see any warnings or errors in my local windows build or the ubuntu CI. I'll change it as you suggest

The warnings could be seen at https://github.com/python/cpython/pull/21378/files I didn't check which CI emitted these warnings.

Reply:
@vstinner I see. I'm not used to checking the files for compiler output.